.github/workflows/ci.yml

```diff
@@ -0,0 +1,175 @@
+name: EFCore.PG CI
+
+on: push
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+        dotnet: [ '3.0.100' ]
+        pg_major: [ 11, 10 ]
+        include:
+          - pg_major: 11
+            edb_version: '11.4-2'
+            postgis_version: '2.5.3'
+          - pg_major: 10
+            edb_version: '10.9-2'
+            postgis_version: '2.5.3'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Start PostgreSQL ${{ matrix.pg_major }} (docker)
+        if: startsWith(matrix.os, 'ubuntu')
+        env:
+          IMAGE: npgsql/postgres:${{ matrix.pg_major }}
+        run: docker run --detach --name postgresql -p 5432:5432 $IMAGE
+        shell: bash
+
+      - name: Start PostgreSQL ${{ matrix.pg_major }} (install)
+        if: startsWith(matrix.os, 'windows')
+        env:
+          PG_MAJOR: ${{ matrix.pg_major }}
+          EDB_VERSION: ${{ matrix.edb_version }}
+          POSTGIS_VERSION: ${{ matrix.postgis_version }}
+        run: |
+          # Install PostgreSQL
+          curl -o pgsql.zip -L https://get.enterprisedb.com/postgresql/postgresql-${EDB_VERSION}-windows-x64-binaries.zip
+          unzip pgsql.zip -x 'pgsql/include/**' 'pgsql/doc/**' 'pgsql/pgAdmin 4/**' 'pgsql/StackBuilder/**'
+
+          # Match Npgsql CI Docker image and stash one level up
+          cp {$GITHUB_WORKSPACE/.build/docker,pgsql}/server.crt
+          cp {$GITHUB_WORKSPACE/.build/docker,pgsql}/server.key
+
+          # Install PostGIS
+          POSTGIS_FILE="postgis-bundle-pg${PG_MAJOR}-${POSTGIS_VERSION}x64"
+          curl -o postgis.zip -L https://download.osgeo.org/postgis/windows/pg${PG_MAJOR}/${POSTGIS_FILE}.zip
+          unzip postgis.zip -d postgis
+          cp -a postgis/$POSTGIS_FILE/. pgsql/
+
+          # Start PostgreSQL
+          pgsql/bin/initdb -D pgsql/PGDATA -E UTF8 -U postgres
+          pgsql/bin/pg_ctl -D pgsql/PGDATA -l logfile -o '-c max_prepared_transactions=10 -c ssl=true -c ssl_cert_file=../server.crt -c ssl_key_file=../server.key' start
+
+          # Configure test account
+          pgsql/bin/psql -U postgres -c "CREATE ROLE npgsql_tests SUPERUSER LOGIN PASSWORD 'npgsql_tests'"
+          pgsql/bin/psql -U postgres -c "CREATE DATABASE npgsql_tests OWNER npgsql_tests"
+        shell: bash
+
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v1.0.2 # Issues with DOTNET_ROOT
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
```

Comment:
Interesting, I'm not sure I see us doing a matrix test of dotnet *SDKs*.... I don't see what value this would actually add, as it has no effect on the actual TFMs we're targeting etc...

What do you have in mind? Should we remove it and just hard-code to 3.0.0? If we do so we may be able to remove this step altogether since it will soon be bundled [in the Github Actions image](https://help.github.com/en/articles/software-in-virtual-environments-for-github-actions#ubuntu-1804-lts) (although we'll need it again if we want to use previews again, which we probably will).

Reply:
I think we're on the same page. I'm using the matrix here just to centralize software versions at the top of the job, but only anticipate it ever having one version at a time. Based on the documentation, there don't seem to be a way yet to set workflow constant variables, so the matrix was the closest I could find.

Reply:
OK. I'd still rather not using the matrix as a variable mechanism - it shows up in the github UI as another matrix parameter (which it isn't really), etc.

I actually don't mind just putting the version inline. Another option which should work is to define an environment variable at the job level (see [jobs.<job_id>.env](https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idenv)).

Reply:
Can't believe I missed that, must've read that page a dozen times.

Reply:
I tried a few different syntaxes, but couldn't get a top-level or job-level variable to resolve in the `with` block of an action. It would be nice to parameterize the SDK version, but it's not important enough to block on, so just inlining it for now as you suggested.